packages/common/test/index.test.js

```diff
@@ -866,95 +867,107 @@ describe('validate', () => {
   });
 
   it('should clear the cursor', () => {
     const state = {
-      cursor: new Date()
-    }
-    const result = cursor()(state)
-    expect(result.cursor).to.eql(undefined)
+      cursor: new Date(),
+    };
+    const result = cursor()(state);
+    expect(result.cursor).to.eql(undefined);
   });
 
   it('should use a default value', () => {
-    const state = {}
-    const result = cursor(state.cursor, { defaultValue: 33 })(state)
-    expect(result.cursor).to.eql(33)
+    const state = {};
+    const result = cursor(state.cursor, { defaultValue: 33 })(state);
+    expect(result.cursor).to.eql(33);
   });
 
   it('should use a custom key', () => {
-    const state = {}
-    const result = cursor(44, { key: 'page' })(state)
-    expect(result.page).to.eql(44)
+    const state = {};
+    const result = cursor(44, { key: 'page' })(state);
+    expect(result.page).to.eql(44);
   });
 
   it('should re-use a custom key', () => {
-    const state = {}
-    const result1 = cursor(44, { key: 'page' })(state)
-    expect(result1.page).to.eql(44)
+    const state = {};
+    const result1 = cursor(44, { key: 'page' })(state);
+    expect(result1.page).to.eql(44);
 
-    const result2 = cursor(55)(state)
-    expect(result2.page).to.eql(55)
+    const result2 = cursor(55)(state);
+    expect(result2.page).to.eql(55);
   });
 
   // testing the log output is hard here, I've only verified it manally
   it('should use an object', () => {
-    const state = {}
-    const c = { page: 22, next: 23 }
-    const result = cursor(c, { key: 'cursor' })(state)
-    expect(result.cursor).to.eql(c)
+    const state = {};
+    const c = { page: 22, next: 23 };
+    const result = cursor(c, { key: 'cursor' })(state);
+    expect(result.cursor).to.eql(c);
   });
 
   it('should apply a custom formatter', () => {
     const state = {};
     const result = cursor('abc', {
-      format: (c) => c.toUpperCase()
+      format: c => c.toUpperCase(),
     })(state);
-    expect(result.cursor).to.eql('ABC')
-  })
+    expect(result.cursor).to.eql('ABC');
+  });
 
   it('should format "today"', () => {
     const state = {};
     const date = new Date().toDateString();
     const result = cursor('today', {
-      format: (c) => c.toDateString()
+      format: c => c.toDateString(),
     })(state);
-    expect(result.cursor).to.eql(date)
-  })
+    expect(result.cursor).to.eql(date);
+  });
 
   it('should format a number to an arbitrary object', () => {
     const state = {};
     const result = cursor(3, {
-      format: (c) => ({ page: c })
+      format: c => ({ page: c }),
     })(state);
-    expect(result.cursor).to.eql({ page: 3 })
-  })
+    expect(result.cursor).to.eql({ page: 3 });
+  });
+});
 
-});
\ No newline at end of file
+describe('group', () => {
+  it('should group an array of objects by a specified key path. ', async function () {
+    const state = {};
+    let resultingState = await group(testData.store.book, 'category')(state);
```

Comment:
There's no need to `await` this - it's a bit misleading. This is a synchronous function.

And why `let resultingState`?

Reply:
Let me fix that, it should be a const

Reply:
Fixed here [f92e3c9](https://github.com/OpenFn/adaptors/pull/621/commits/f92e3c98396d673f5036311894c12665db415461)